src/main/java/core/basesyntax/service/DataConverterImpl.java

```
@@ -0,0 +1,45 @@
+package core.basesyntax.service;
+
+import core.basesyntax.exception.FruitShopException;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.model.Operation;
+import java.util.ArrayList;
+import java.util.List;
+
+public class DataConverterImpl implements DataConverter {
+    private static final String COMMA = ",";
+
+    @Override
+    public List<FruitTransaction> convertToTransaction(List<String> rawLines) {
+        List<FruitTransaction> result = new ArrayList<>();
+        if (rawLines == null || rawLines.isEmpty()) {
+            return result;
```

convertToTransaction returns an empty list when rawLines == null (lines 15–16). This violates the requirement to validate null parameters. From the task checklist (DataConverter requirements):
"3.3. Implement a DataConverter/Parser service that converts List<String> to List<FruitTransaction: - Skip or correctly handle header line. - Validate that each data line has exactly 3 comma-separated values: type, fruit, quantity. - Validate operation code is one of b, s, p, r; unknown codes should cause an error. - Validate fruit name is non-null/non-empty. - Validate quantity is an integer and >= 0; negative quantities should cause an error."
You must throw a FruitShopException when rawLines is null (do not silently return an empty list).